stable_plugins/classical_ml/data_preparation/sym_max_mean.py

```diff
@@ -357,47 +355,56 @@ def calculation_task(self, db_id: int) -> str:
                 if ent_attr1 is None or ent_attr2 is None:
                     sym_max_mean = None  # TODO: add handling of missing values
                 else:
+                    if isinstance(ent_attr1, set):
+                        ent_attr1 = list(ent_attr1)
+
+                    if isinstance(ent_attr2, set):
+                        ent_attr2 = list(ent_attr2)
+
                     if not isinstance(ent_attr1, list):
                         ent_attr1 = [ent_attr1]
 
                     if not isinstance(ent_attr2, list):
                         ent_attr2 = [ent_attr2]
 
-                    # calculate Sym Max Mean
+                    if len(ent_attr1) == 0 or len(ent_attr2) == 0:
+                        sym_max_mean = 0
```

Comment:
Should this also apply if one side is None? (and if both are None/empty, should they have a similarity > 0 or even 1?) Maybe those default values could even be parameters in the frontend.

Reply:
If both are empty/None, then None is the correct value, if only one side is empty/None, than 0 should be safe (they are not similar in that case).
How is None handled by the next in line plugin? Maybe that could reasonably assume None==0?

Reply:
The "similarity to distances" plugin currently doesn't handle None values. I think it should just output None as well. The plugin after that would be the aggregator. I think it makes sense for this plugin to handle None values.